# Ejercicio 1

# Minimizar el costo de los suplementos Vega Vita y Happy Health

In [1]:
import pulp as pp


Queremos minimizar el costo de los suplementos.

In [50]:
model = pp.LpProblem(name='supplement-problem', sense=pp.LpMinimize)

In [52]:
VegaVita = pp.LpVariable(name="VegaVita", lowBound=0, cat='Continuous')
HappyHealth = pp.LpVariable(name="HappyHealth", lowBound=0, cat='Continuous')

Definir la Función Objetivo

In [54]:
obj_func = 0.2 * VegaVita + 0.3 * HappyHealth


Vitamina C: 20 mg por pastilla de Vega Vita, 30 mg por pastilla de Happy Health. Necesitamos al menos 60 mg.

In [57]:
C1 = pp.LpConstraint(name='Vitamin_C_Constraint', e=20 * VegaVita + 30 * HappyHealth, rhs=60, sense=pp.LpConstraintGE)


Calcio: 500 mg por pastilla de Vega Vita, 250 mg por pastilla de Happy Health. Necesitamos al menos 1000 mg.

In [60]:
C2 = pp.LpConstraint(name='Calcium_Constraint', e=500 * VegaVita + 250 * HappyHealth, rhs=1000, sense=pp.LpConstraintGE)


Hierro: 9 mg por pastilla de Vega Vita, 2 mg por pastilla de Happy Health. Necesitamos al menos 18 mg.

In [63]:
C3 = pp.LpConstraint(name='Iron_Constraint', e=9 * VegaVita + 2 * HappyHealth, rhs=18, sense=pp.LpConstraintGE)


Niacina: 2 mg por pastilla de Vega Vita, 10 mg por pastilla de Happy Health. Necesitamos al menos 20 mg.

In [66]:
C4 = pp.LpConstraint(name='Niacin_Constraint', e=2 * VegaVita + 10 * HappyHealth, rhs=20, sense=pp.LpConstraintGE)


Magnesio: 60 mg por pastilla de Vega Vita, 90 mg por pastilla de Happy Health. Necesitamos al menos 360 mg.

In [69]:
C5 = pp.LpConstraint(name='Magnesium_Constraint', e=60 * VegaVita + 90 * HappyHealth, rhs=360, sense=pp.LpConstraintGE)


In [71]:
model += obj_func
model += C1
model += C2
model += C3
model += C4
model += C5


In [73]:
solverToUse = pp.PULP_CBC_CMD(msg=False)
model.solve(solver=solverToUse)

1

In [75]:
import pandas as pd
Results = {"Model Status": pp.LpStatus[model.status]}
Results.update({"Optimal Solution": pp.value(model.objective)})
Results.update({v.name: v.varValue for v in model.variables()})
pd.DataFrame.from_dict(Results, orient='index').T.set_index('Model Status').style.format('{:,.2f}')

El resultado indica que se debe combinar ambos suplementos para satisfacer las necesidades nutricionales de la manera más económica posible. En este caso, debes adquirir 3.13 tabletas de Happy Health y 1.30 tabletas de Vega Vita. Dado que los suplementos normalmente vienen en cantidades enteras, puede ser necesario redondear los valores 

# Redondeo

In [79]:
VegaVita = pp.LpVariable(name="VegaVita", lowBound=0, cat='Integer')
HappyHealth = pp.LpVariable(name="HappyHealth", lowBound=0, cat='Integer')


In [81]:
model = pp.LpProblem(name='supplement-problem', sense=pp.LpMinimize)


In [83]:
VegaVita = pp.LpVariable(name="VegaVita", lowBound=0, cat='Integer')
HappyHealth = pp.LpVariable(name="HappyHealth", lowBound=0, cat='Integer')


In [85]:
obj_func = 0.2 * VegaVita + 0.3 * HappyHealth


In [87]:
C1 = pp.LpConstraint(name='Vitamin_C_Constraint', e=20 * VegaVita + 30 * HappyHealth, rhs=60, sense=pp.LpConstraintGE)
C2 = pp.LpConstraint(name='Calcium_Constraint', e=500 * VegaVita + 250 * HappyHealth, rhs=1000, sense=pp.LpConstraintGE)
C3 = pp.LpConstraint(name='Iron_Constraint', e=9 * VegaVita + 2 * HappyHealth, rhs=18, sense=pp.LpConstraintGE)
C4 = pp.LpConstraint(name='Niacin_Constraint', e=2 * VegaVita + 10 * HappyHealth, rhs=20, sense=pp.LpConstraintGE)
C5 = pp.LpConstraint(name='Magnesium_Constraint', e=60 * VegaVita + 90 * HappyHealth, rhs=360, sense=pp.LpConstraintGE)

In [89]:
model += obj_func
model += C1
model += C2
model += C3
model += C4
model += C5

In [91]:
solverToUse = pp.PULP_CBC_CMD(msg=False)
model.solve(solver=solverToUse)

1

In [93]:
Results = {"Model Status": pp.LpStatus[model.status]}
Results.update({"Optimal Solution": pp.value(model.objective)})
Results.update({v.name: v.varValue for v in model.variables()})
pd.DataFrame.from_dict(Results, orient='index').T.set_index('Model Status').style.format('{:,.2f}')

Se obtiene una solución óptima que minimiza el costo a 1.20 dólares diarios, cumpliendo con todos los requisitos nutricionales. Para lograrlo, se deben consumir 3 tabletas de Happy Health y 3 tabletas de Vega Vita al día, garantizando una solución práctica donde las cantidades de tabletas son enteras. Este ajuste asegura que los requerimientos de vitamina C, calcio, hierro, niacina y magnesio se cumplen sin la necesidad de fraccionar suplementos, lo cual no es práctico en la vida real. Aunque el costo no cambió significativamente respecto al modelo anterior, ahora los resultados son más practicos.

# Ejercicio 1.A

# The scheduling problem

In [6]:
import pulp as pp


In [8]:
model = pp.LpProblem(name="bus_scheduling_problem", sense=pp.LpMinimize)


In [10]:
DR_0_8 = pp.LpVariable(name="DR_0_8", lowBound=0, cat='Integer')
DR_4_12 = pp.LpVariable(name="DR_4_12", lowBound=0, cat='Integer')
DR_8_16 = pp.LpVariable(name="DR_8_16", lowBound=0, cat='Integer')
DR_12_20 = pp.LpVariable(name="DR_12_20", lowBound=0, cat='Integer')
DR_16_0 = pp.LpVariable(name="DR_16_0", lowBound=0, cat='Integer')
DR_20_4 = pp.LpVariable(name="DR_20_4", lowBound=0, cat='Integer')


Se definen seis variables (de tipo entero) que representan el número de conductores asignados a distintos turnos. Cada variable tiene un límite inferior (lowBound) de 0, es decir, no puede tomar valores negativos.

Las variables tienen los siguientes nombres:

DR_0_8: Conductores asignados de 0:00 a 8:00 horas.

DR_4_12: Conductores asignados de 4:00 a 12:00 horas.

DR_8_16: Conductores asignados de 8:00 a 16:00 horas.

DR_12_20: Conductores asignados de 12:00 a 20:00 horas.

DR_16_0: Conductores asignados de 16:00 a 0:00 horas.

DR_20_4: Conductores asignados de 20:00 a 4:00 horas.

In [12]:
model += (DR_0_8 + DR_4_12 + DR_8_16 + DR_12_20 + DR_16_0 + DR_20_4, "Total_Drivers")


Esta línea define la función objetivo que se debe minimizar. En este caso, la función objetivo es la suma total de conductores asignados a todos los turnos, buscando minimizar el número total de conductores.

Restricciones:

In [14]:
model += (DR_0_8 + DR_20_4 >= 4, "Demand_0_4")


In [16]:
model += (DR_0_8 + DR_4_12 >= 8, "Demand_4_8")


In [18]:
model += (DR_4_12 + DR_8_16 >= 10, "Demand_8_12")


In [20]:
model += (DR_8_16 + DR_12_20 >= 7, "Demand_12_16")


In [22]:
model += (DR_12_20 + DR_16_0 >= 12, "Demand_16_20")


model += (DR_16_0 + DR_20_4 >= 4, "Demand_20_24")


In [26]:
solver = pp.PULP_CBC_CMD(msg=False)
model.solve(solver=solver)


1

In [28]:
for var in [DR_0_8, DR_4_12, DR_8_16, DR_12_20, DR_16_0, DR_20_4]:
    print(f"{var.name}: {var.varValue}")

print(f"Total Drivers: {pp.value(model.objective)}")


DR_0_8: 0.0
DR_4_12: 10.0
DR_8_16: 0.0
DR_12_20: 12.0
DR_16_0: 0.0
DR_20_4: 4.0
Total Drivers: 26.0


Interpretación de los Resultados


DR_0_8: 0.0: No se asignan conductores para el turno de 0:00 a 8:00 horas

DR_4_12: 10.0: Se asignan 10 conductores para el turno de 4:00 a 12:00 horas.

DR_8_16: 0.0: No se asignan conductores para el turno de 8:00 a 16:00 horas.

DR_12_20: 12.0: Se asignan 12 conductores para el turno de 12:00 a 20:00 horas.

DR_16_0: 0.0: No se asignan conductores para el turno de 16:00 a 0:00 horas.

DR_20_4: 4.0: Se asignan 4 conductores para el turno de 20:00 a 4:00 horas.

El objetivo del problema era minimizar el número de conductores necesarios para cubrir las necesidades de transporte durante las 24 horas del día, respetando ciertas demandas mínimas en cada tramo horario. La solución encontró que con 26 conductores se pueden cumplir todas las restricciones establecidas, distribuidos de manera específica según los requerimientos de cada turno.

# Ejercicio 2

In [34]:
file_path = r"C:\Users\Sistema\OneDrive\Documentos\PUCP\2024-2\Herramientas cuantitativas\Tarea 5\Optimizaci-n\comparissons.xlsx"

In [36]:
import pandas as pd

pairwise_cost = pd.read_excel(file_path, sheet_name='cost_liv', index_col=0)
pairwise_language = pd.read_excel(file_path, sheet_name='language_diff', index_col=0)
pairwise_job = pd.read_excel(file_path, sheet_name='job_poss', index_col=0)
pairwise_criteria = pd.read_excel(file_path, sheet_name='criteria', index_col=0)


In [38]:
print("Matriz de Comparación - Costo de Vida:")
print(pairwise_cost)
print("\nMatriz de Comparación - Dificultad del Idioma:")
print(pairwise_language)
print("\nMatriz de Comparación - Oportunidades de Empleo:")
print(pairwise_job)
print("\nMatriz de Comparación - Criterios:")
print(pairwise_criteria)

Matriz de Comparación - Costo de Vida:
         Brazil     Spain       USA  Germany
Brazil        1  0.166667  0.142857    0.125
Spain         6  1.000000  0.333333    0.250
USA           7  3.000000  1.000000    0.500
Germany       8  4.000000  2.000000    1.000

Matriz de Comparación - Dificultad del Idioma:
         Brazil  Spain       USA   Germany
Brazil        1    0.2  0.142857  0.111111
Spain         5    1.0  0.333333  0.200000
USA           7    3.0  1.000000  0.333333
Germany       9    5.0  3.000000  1.000000

Matriz de Comparación - Oportunidades de Empleo:
         Brazil     Spain   USA   Germany
Brazil        1  0.333333  0.20  0.250000
Spain         3  1.000000  0.25  0.333333
USA           5  4.000000  1.00  0.500000
Germany       4  3.000000  2.00  1.000000

Matriz de Comparación - Criterios:
               cost_liv  language_diff  job_poss
cost_liv            1.0       0.333333         2
language_diff       3.0       1.000000         4
job_poss            0.5       

In [40]:
import networkx as nx

In [42]:
def get_comparisons_from_matrix(matrix):
    G = nx.from_pandas_adjacency(matrix, create_using=nx.MultiDiGraph())
    return {(e[0], e[1]): e[2]['weight'] for e in G.edges(data=True) if e[0] != e[1]}

In [44]:
cost_comparisons = get_comparisons_from_matrix(pairwise_cost)
language_comparisons = get_comparisons_from_matrix(pairwise_language)
job_comparisons = get_comparisons_from_matrix(pairwise_job)
criteria_comparisons = get_comparisons_from_matrix(pairwise_criteria)

In [46]:
import ahpy
cost_l = ahpy.Compare('Cost of Living', cost_comparisons, precision=3, random_index='saaty')
language_d = ahpy.Compare('Language Difficulty', language_comparisons, precision=3, random_index='saaty')
job_p = ahpy.Compare('Job Opportunities', job_comparisons, precision=3, random_index='saaty')

In [48]:
criteria = ahpy.Compare('Criteria', criteria_comparisons, precision=3, random_index='saaty')
criteria.add_children([cost_l, language_d, job_p])


In [50]:
import numpy as np

def ahp_weights_and_consistency(matrix):
    eigenvalues, eigenvectors = np.linalg.eig(matrix)
    max_eigenvalue = np.max(eigenvalues)
    weights = eigenvectors[:, np.argmax(eigenvalues)].real
    weights = weights / weights.sum()

    n = matrix.shape[0]
    CI = (max_eigenvalue - n) / (n - 1)
    RI = [0, 0, 0.58, 0.9, 1.12, 1.24, 1.32, 1.41, 1.45]  # Valores RI típicos para matrices de 1 a 9
    CR = CI / RI[n - 1] if n - 1 < len(RI) else None

    return weights, CI, CR


In [56]:
pip install --upgrade ahpy


Note: you may need to restart the kernel to use updated packages.


In [60]:
cost_weights, cost_CI, cost_CR = ahp_weights_and_consistency(pairwise_cost.values)
language_weights, language_CI, language_CR = ahp_weights_and_consistency(pairwise_language.values)
job_weights, job_CI, job_CR = ahp_weights_and_consistency(pairwise_job.values)
criteria_weights, criteria_CI, criteria_CR = ahp_weights_and_consistency(pairwise_criteria.values)



In [68]:
criteria_weights = criteria_weights  # Este debe ser el vector de pesos de los criterios


cost_of_living_weights = cost_weights
language_difficulty_weights = language_weights
job_opportunities_weights = job_weights

countries = ['Brazil', 'Spain', 'USA', 'Germany']

final_scores = {}
for i, country in enumerate(countries):
    final_score = (
        criteria_weights[0] * cost_of_living_weights[i] +  # Costo de Vida
        criteria_weights[1] * language_difficulty_weights[i] +  # Dificultad del Idioma
        criteria_weights[2] * job_opportunities_weights[i]  # Oportunidades de Empleo
    )
    final_scores[country] = final_score

sorted_final_scores = dict(sorted(final_scores.items(), key=lambda item: item[1], reverse=True))

print("Resultados ordenados:")
print(sorted_final_scores)


Resultados ordenados:
{'Germany': 0.53140420939152, 'USA': 0.28992510351727213, 'Spain': 0.1341833331561339, 'Brazil': 0.044487353935074046}


El análisis AHP se utilizó para determinar el país más adecuado para un programa de maestría, evaluando tres criterios principales: Costo de Vida, Dificultad del Idioma, y Oportunidades de Empleo. Los países considerados en este análisis fueron Brazil, Spain, USA, y Germany. Cada criterio se ponderó en función de su importancia relativa, y luego se evaluaron los países en cada criterio.

Los resultados finales muestran el siguiente ranking de países, basado en el puntaje global calculado:

Germany: 0.5314 USA: 0.2899 Spain: 0.1342 Brazil: 0.0445 Este ranking indica que Germany es la opción más adecuada para el programa de maestría, ya que tiene el puntaje global más alto. Esto sugiere que Alemania se destaca en los aspectos clave que se consideraron, particularmente en términos de Dificultad del Idioma y Oportunidades de Empleo después de la maestría. USA ocupa el segundo lugar, siendo también una opción competitiva pero menos favorable en comparación con Alemania. Spain y Brazil quedan en las posiciones tercera y cuarta, lo que implica que, aunque podrían ser opciones viables, no cumplen tan efectivamente con los criterios considerados en comparación con los otros dos países.

En conclusión, Germany es la opción recomendada según los criterios seleccionados, ofreciendo un equilibrio favorable entre el costo de vida, la facilidad del idioma y las oportunidades de empleo al finalizar los estudios.

# Ejercicio 3 - Benchmarking para el programa Pensión 65 por departamento en 2024

Se realizará el ejercicio de Benchmarking en base a la información acerca del programa Pensión 65 por departamento ya que no existe información por provincia o por distrito. Se recopiló la información a través del portal de Transparencia Económica y los datos estadísticos del programa. Las variables elegidas son las siguientes:

- **departamento**: Se refiere al departamento del Perú.
- **puntos_pago**: Se refiere al número de puntos de pago establecidos para el programa Pensión 65 para el 2024.
- **presupuesto_p65**: Se refiere al Presupuesto Institucional Modificado (PIM) para el programa Pensión 65 para el 2024.
- **transferencia_p65**: Se refiere al dinero transferido a los usuarios bimestralmente en el programa Pensión 65 del último bimestre.
- **atendidos_p65**: Se refiere a la cantidad de usuarios atendidos en el último bimestre en el programa Pensión 65.

Las variables puntos_pago y presupuesto_p65 son las variables *input* mientras que las variables transferencia_p65 y atendidos_p65 son las variables *output*. En base a esto se realizarán las mediciones de eficiencia. 

In [102]:
import pandas as pd
import warnings

warnings.filterwarnings("ignore")
url = 'https://docs.google.com/spreadsheets/d/142EE3vg79P9LWelDsGlNfFeJwxdndgdh152rcJGb9Yc/export?format=csv'

p65 = pd.read_csv(url)
print(p65.head())

  departamento  puntos_pago  presupuesto_p65  transferencia_p65  atendidos_p65
0     Amazonas           35         33016281            5170250          20681
1       Ancash           72         75956506           12135000          48540
2     Apurímac           82         53786680            8303500          33214
3     Arequipa           42         25284352            3851000          15404
4     Ayacucho           60         65903382           10406750          41627


Se crean los ratios para realizar las mediciones de eficiencia y se grafican los ratios.

In [105]:
p65['rate_AtendidosByPuntos']=(p65.atendidos_p65/p65.puntos_pago)
p65['rate_TransferenciaByPresupuesto']=(p65.transferencia_p65/p65.presupuesto_p65)

In [107]:
import altair as alt

points = alt.Chart(p65).mark_point().encode(
    x='rate_AtendidosByPuntos:Q',
    y='rate_TransferenciaByPresupuesto:Q',
    color=alt.Color('departamento', legend=alt.Legend(titleFontSize=12, labelFontSize=10))  # Tamaño más pequeño para la leyenda
).properties(
    width=600,  
    height=600  
)

text = points.mark_text(
    align='right',
    baseline='middle',
    dx=-7,
    size=7  
).encode(
    text='departamento'
)

(points + text).interactive()

alt.LayerChart(...)

Se observan los departamentos y se comparan los ratios. Parece ser que la eficiencia en los ratios no es igual en los departamentos. 

In [109]:
p65[['departamento','rate_AtendidosByPuntos','rate_TransferenciaByPresupuesto']].sort_values(by='rate_AtendidosByPuntos',ascending=False).head()

departamento  rate_AtendidosByPuntos  rate_TransferenciaByPresupuesto
23       Tumbes             1712.000000                         0.155047
24      Ucayali             1590.750000                         0.160502
6        Callao             1431.200000                         0.169468
11        Junín             1328.833333                         0.161983
21   San Martín             1240.586207                         0.161083

In [114]:
p65[['departamento','rate_AtendidosByPuntos','rate_TransferenciaByPresupuesto']].sort_values(by='rate_TransferenciaByPresupuesto',ascending=False).head()

departamento  rate_AtendidosByPuntos  rate_TransferenciaByPresupuesto
6        Callao             1431.200000                         0.169468
13   Lambayeque             1078.240000                         0.162189
19        Piura             1156.642857                         0.162114
11        Junín             1328.833333                         0.161983
20         Puno             1011.287671                         0.161944

In [116]:
Best_AtendidosByPuntos=p65.rate_AtendidosByPuntos.idxmax()
Best_TransferenciaByPresupuesto=p65.rate_TransferenciaByPresupuesto.idxmax()

frontier1=p65.loc[Best_AtendidosByPuntos,['rate_AtendidosByPuntos','rate_TransferenciaByPresupuesto']].to_list()
frontier2=p65.loc[Best_TransferenciaByPresupuesto,['rate_AtendidosByPuntos','rate_TransferenciaByPresupuesto']].to_list()

#parallels
frontier1v=[frontier1[0],0]
frontier2h=[0,frontier2[1]]

#then
envelope=pd.DataFrame([frontier2h,frontier2,frontier1,frontier1v],columns=['x','y'])
envelope

x         y
0     0.0  0.169468
1  1431.2  0.169468
2  1712.0  0.155047
3  1712.0  0.000000

In [118]:
points + text + alt.Chart(envelope).mark_line(color='red').encode(
    x='x',
    y='y',
)

alt.LayerChart(...)

Debido a que hay múltiples inputs y outputs, se utilizará Pyfrontier para desarrollar la optimización

In [121]:
!pip install Pyfrontier

In [123]:
p65Input = ['puntos_pago', 'presupuesto_p65']  
p65Output = ['transferencia_p65', 'atendidos_p65'] 

from Pyfrontier.frontier_model import EnvelopDEA

dea_p65_vrs_in = EnvelopDEA("VRS", "in")

dea_p65_vrs_in.fit(
    inputs=p65[p65Input].to_numpy(),
    outputs=p65[p65Output].to_numpy()
)


In [125]:
p65['vrs_in']=[r.score for r in dea_p65_vrs_in.result]
p65.set_index(p65.departamento,inplace=True)
p65['vrs_in']

departamento
Amazonas         0.954869
Ancash           0.983719
Apurímac         0.947407
Arequipa         0.923281
Ayacucho         0.971144
Cajamarca        1.000000
Callao           1.000000
Cusco            0.990195
Huancavelica     0.951398
Huánuco          0.974999
Ica              0.936712
Junín            1.000000
La Libertad      0.982689
Lambayeque       0.992897
Lima             0.973458
Loreto           0.961973
Madre de Dios    1.000000
Moquegua         0.913017
Pasco            0.916234
Piura            1.000000
Puno             1.000000
San Martín       0.991400
Tacna            0.944856
Tumbes           1.000000
Ucayali          1.000000
Name: vrs_in, dtype: float64

Después de realizar el ejercicio de DEA, se puede observar la eficiencia de los departamentos en el marco del programa social Pensión 65. Como se puede observar, todos los departamentos tienen una eficiencia superior al 90%. Al considerar los inputs y outputs seleccionados para realizar el análisis, es posible afirmar que la eficiencia de los puntos de pagos para atender a los usuarios del programa y el uso del presupuesto en transferencias para los usuarios es alta. Sin embargo, se sabe que el programa tiene varios problemas en el reconocimiento de usuarios, el alcance de los puntos de pago y brindar información a la población atendida. Sería interesante realizar el análisis con información mucho más exacta y extensa acerca de la ejecución e implementación del programa para considerar todos los aspectos que puedan intervenir en la medición de su eficiencia.

Por otra parte, en la tabla inferior, no es posible dar una interpretación correcta sobre los resultados. Como puede observarse, no ha sido posible calcular algunos coeficientes para la interpretación del modelo. Probablemente se deba a problemas con la muestra, la colinalidad entre las independientes, entre otros. Se debería observar el error para desarrollar un nuevo modelo que incluya otras variables y expandir la muestra a provincias o distritos, si se obtiene la información

In [128]:
import numpy as np
from py4etrics import tobit

# Definir la variable censurada en base a vrs_in
p65['censored'] = np.where(p65['vrs_in'] == 1, 1, 0)

# Variables censurada, dependiente e independientes
cens = p65['censored']
endog = p65['vrs_in']  # Variable dependiente
exog = p65[['puntos_pago', 'presupuesto_p65']]  # Variables independientes (ajustadas a tus datos)

# Aplicar la regresión Tobit
tobit_res = tobit.Tobit(endog, exog, cens, right=1).fit()

# Mostrar resumen de resultados
tobit_res.summary()

Optimization terminated successfully.
         Current function value: 0.886631
         Iterations: 64
         Function evaluations: 115


C:\Users\Sistema\anaconda3\Lib\site-packages\statsmodels\base\model.py:595: HessianInversionWarning: Inverting hessian failed, no bse or cov_params available
  warnings.warn('Inverting hessian failed, no bse or cov_params '


<class 'statsmodels.iolib.summary.Summary'>
"""
                              Tobit Regression Results                             
===================================================================================
Dep. Variable:                      vrs_in   Pseudo R-squ:                    1.924
Method:                 Maximum Likelihood   Log-Likelihood:                  -22.2
No. Observations:                       25   LL-Null:                          24.0
No. Uncensored Obs:                     17   LL-Ratio:                        -92.3
No. Left-censored Obs:                   0   LLR p-value:                     1.000
No. Right-censored Obs:                  8   AIC:                              46.3
Df Residuals:                           23   BIC:                              47.6
Df Model:                                1   Covariance Type:             nonrobust
===================================================================================
                      coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------
puntos_pago         0.0016        nan        nan        nan         nan         nan
presupuesto_p65  1.604e-08        nan        nan        nan         nan         nan
Log(Sigma)         -0.4558        nan        nan        nan         nan         nan
===================================================================================
"""